src/ui/components/filters/sortoptionscomponent.js

```diff
@@ -23,31 +23,53 @@ export default class SortOptionsComponent extends Component {
     this.selectedOptionIndex = parseInt(this.core.globalStorage.getState(this.name)) || 0;
     this.options[this.selectedOptionIndex].isSelected = true;
     this.hideExcessOptions = this._config.showMore && this.selectedOptionIndex < this._config.showMoreLimit;
-    this.showReset = this._config.showReset && this.selectedOptionIndex !== 0;
+    this.searchOnChangeIsEnabled = this._config.searchOnChange;
+    this.showResetIsEnabled = this._config.showReset;
+    this.showReset = this.showResetIsEnabled && this.selectedOptionIndex !== 0;
+    this.isNoResults = false;
 
     /**
-     * This component listens to updates to vertical results, and sets its state to it when
-     * an update occurs.
-     * @type {string}
+     * This component should only render if there are search results, so it should listen
+     * to updates to vertical results and handle them accordingly.
      */
     this.core.globalStorage.on('update', StorageKeys.VERTICAL_RESULTS, verticalResults => {
-      if (verticalResults.searchState === SearchStates.SEARCH_COMPLETE) {
-        this.setState(verticalResults);
+      const isSearchComplete = verticalResults.searchState === SearchStates.SEARCH_COMPLETE;
+
+      if (isSearchComplete) {
+        const isNoResults = verticalResults.resultsContext === ResultsContext.NO_RESULTS;
+        this.handleVerticalResultsUpdate(isNoResults);
       }
     });
   }
 
-  setState (data = {}) {
+  handleVerticalResultsUpdate (isNoResults) {
+    const isPreviousSearchResults = !this.isNoResults;
+    this.isNoResults = isNoResults;
+
+    if (isNoResults) {
+      this.setState(); // Call setState to trigger a re-render
+      return;
+    }
+
+    // If previously there weren't any search results, but now there are, re-render by
+    // calling setState. Otherwise, don't re-render in order to maintain focus on the
+    // selected sort option.
+    if (!isPreviousSearchResults) {
```

Comment:
does isPreviousSearchResults and this if block do anything? if isNoResults is true it won't reach this line, and if it's false the if statement will be skipped over

Reply:
If `isNoResults` is false, this if statement isn't skipped over. The purpose of this block is to trigger a re-render when previously there were no results, and now there are. If there previously were results, and there still are results, then we don't want to re-render.

Reply:
oh sorry I read !this.isNoResults as just !isNoResults, lgtm!

Reply:
would `wasNoResults` be clearer than `isPreviousSearchResults`?

Reply:
would it be the same behavior if we just checked whether the isNoResults state changed? instead of 2 if statements

Reply:
ya, that works!